pkg/infrastructure/openstack/postprovision/floatingips.go

```diff
@@ -40,11 +41,24 @@ func FloatingIPs(ctx context.Context, c client.Client, cluster *capo.OpenStackCl
 		return err
 	}
 
-	_, err = createAndAttachFIP(networkClient, "bootstrap", infraID, cluster.Status.ExternalNetwork.ID, bootstrapPort.ID)
+	fip, err := createAndAttachFIP(networkClient, "bootstrap", infraID, cluster.Status.ExternalNetwork.ID, bootstrapPort.ID)
 	if err != nil {
 		return err
 	}
 
+	bootstrapMachine.Status.Addresses = append(bootstrapMachine.Status.Addresses, apiv1.NodeAddress{
+		Type:    apiv1.NodeExternalIP,
+		Address: fip.FloatingIP,
+	})
+
+	// Here Installer is updating the CAPO manifest directly. This is a
+	// hack to make the gather machinery aware of the bootstrap external
+	// IP. We want to remove this whole post-provision step in 4.17 once we
+	// directly pass CAPO a bootstrap manifest requesting a FIP.
+	if err := c.Update(ctx, bootstrapMachine); err != nil {
```

Comment:
I think this will be a no-op because you need to update the status subresource, not the main resource.

```suggestion
	if err := c.Status().Update(ctx, bootstrapMachine); err != nil {
```